driver_test.go

```diff
@@ -1485,6 +1485,24 @@ func TestPodmanDriver_ReadOnlyFilesystem(t *testing.T) {
 	require.True(t, inspectData.HostConfig.ReadonlyRootfs)
 }
 
+// check userns mode configuration (single value)
+func TestPodmanDriver_UsernsMode(t *testing.T) {
+	taskCfg := newTaskConfig("", busyboxLongRunningCmd)
+	taskCfg.UserNS = "host"
+	inspectData := startDestroyInspect(t, taskCfg, "userns")
+
+	require.Equal(t, "host", inspectData.HostConfig.UsernsMode)
+}
+
+// check userns mode configuration (parsed value)
+func TestPodmanDriver_UsernsModeParsed(t *testing.T) {
+	taskCfg := newTaskConfig("", busyboxLongRunningCmd)
+	taskCfg.UserNS = "keep-id:uid=200,gid=210"
+	inspectData := startDestroyInspect(t, taskCfg, "userns")
+
+	require.Equal(t, "keep-id:uid=200,gid=210", inspectData.HostConfig.UsernsMode)
+}
```

Comment:
I'm not sure why the Github Action didn't run (it does this some times 😞), but I ran it locally and I think these need to run with Podman in rootless mode? The first test always returns an empty string:

```
--- FAIL: TestPodmanDriver_UsernsMode (6.28s)
    driver_test.go:1494:
                Error Trace:    /Users/laoqui/go/src/github.com/hashicorp/nomad-driver-podman/driver_test.go:1494
                Error:          Not equal:
                                expected: "host"
                                actual  : ""

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -host
                                +
                Test:           TestPodmanDriver_UsernsMode
FAIL
FAIL    github.com/hashicorp/nomad-driver-podman        6.313s
FAIL
```

The second test says the mode is not supported:

```
--- FAIL: TestPodmanDriver_UsernsModeParsed (4.11s)
    driver_test.go:2253:
                Error Trace:    /Users/laoqui/go/src/github.com/hashicorp/nomad-driver-podman/driver_test.go:2253
                                                        /Users/laoqui/go/src/github.com/hashicorp/nomad-driver-podman/driver_test.go:1501
                Error:          Received unexpected error:
                                rpc error: code = Unknown desc = failed to start task, could not create container: cannot create container, status code: 500: {"cause":"keep-id is only supported in rootless mode","message":"keep-id is only supported in rootless mode","response":500}
                Test:           TestPodmanDriver_UsernsModeParsed
```

I think our tests run as `root` so we may need to skip these in CI.

Reply:
Part of the problem is I'm running a Macbook M1 so let me figure out...

Reply:
Same 😅 

I've been using [Lima](https://github.com/lima-vm/lima) with this configuration, and so far it has been working well:
https://gist.github.com/lgfa29/f24eec7bb39f4bda28d7765bec8740cc#file-podman-yaml

I think you may need to run the tests as `sudo`.